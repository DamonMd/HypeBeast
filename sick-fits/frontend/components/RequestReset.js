import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";

const REQUEST_RESET = gql`
  mutation REQUEST_RESET($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;

class RequestReset extends Component {
  state = {
    email: ""
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  render() {
    return (
      <Mutation mutation={REQUEST_RESET} variables={this.state}>
        {(requestReset, { error, loading, called }) => {
          return (
            <Form
              method="post"
              onSubmit={async e => {
                e.preventDefault();
                const res = await requestReset();
              }}
            >
              <fieldset disabled={loading} aria-disabled={loading}>
                <h2>Request a Password Reset</h2>
                <Error error={error} />
                {!error && !loading && called && (
                  <p>Success! Please check your email for a reset link</p>
                )}
                <label>
                  Email
                  <input
                    name="email"
                    type="email"
                    placeholder="email"
                    value={this.state.email}
                    onChange={this.handleChange}
                  />
                </label>

                <button type="submit">Reset</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default RequestReset;
