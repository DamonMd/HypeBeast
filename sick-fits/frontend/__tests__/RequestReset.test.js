import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import RequestReset, { REQUEST_RESET } from "../components/RequestReset";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";

const mocks = [
  {
    request: {
      query: REQUEST_RESET,
      variables: { email: "person@gmail.com" }
    },
    result: {
      data: { requestReset: { message: "success", __typename: "Message" } }
    }
  }
];

describe("<RequestReset/>", () => {
  it("renders and matches snapshot", () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it("calls the mutation", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    //simulate typing an emal
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "person@gmail.com" }
    });
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    // expect(wrapper.find("p").text()).toContain(
    //   "Success! Please check your email for a reset link"
    // );
  });
});
