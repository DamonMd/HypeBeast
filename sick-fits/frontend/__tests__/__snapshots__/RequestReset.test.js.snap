// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 dMniLL"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-disabled={false}
    disabled={false}
  >
    <h2>
      Request a Password Reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label>
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Reset
    </button>
  </fieldset>
</form>
`;
